### ISSUE TITLE
Distribute `mpm`

### ISSUE THREAD

**Issue by kdeldycke** [kdeldycke/meta-package-manager#527]
To increase and ease `mpm` adoption, we can pre-package it on the most popular package manager for each platform.

See the **[current inventory of distribution channels](https://kdeldycke.github.io/meta-package-manager/benchmark.html#distribution)**.

Some exemples:
- [x] homebrew: https://github.com/Hasnep/homebrew-tap/blob/main/Formula/meta-package-manager.rb
- [ ] chocolatey
- [x] #725 
- [x] Arch Linux: https://aur.archlinux.org/packages/meta-package-manager
- [x] [![Packaging status](https://repology.org/badge/vertical-allrepos/meta-package-manager.svg)](https://repology.org/project/meta-package-manager/versions)

**Comment by autinerd** [comment]
> Oh wow, you rocks! Does that means I can `mpm --pacman install meta-package-manager`?

pacman doesn't support AUR packages, I need to add one of the AUR helpers like yay :smile: I will work on it!

**Comment by kdeldycke** [comment]
I don't know a lot about the Arch Linux ecosystem. But it seems `yay` is a wrapper around `pacman` but allows package installation from the AUR. Am I right? If so then we can subclass the `Pacman` class to add support for `yay` in `mpm`. I can try doing it blindly as it seems both `pacman` and `yay` share the same sets of options.

**Comment by kdeldycke** [comment]
@autinerd I just added installation instruction with `yay` in commit https://github.com/kdeldycke/meta-package-manager/commit/1c6d0ec10a3cf1b039f21771abc9698579846965 .

**Comment by autinerd** [comment]
I also added a AUR package which pulls the git version https://aur.archlinux.org/packages/meta-package-manager-git

**Comment by Hasnep** [comment]
Inspired by @autinerd's Arch package, I've made a homebrew formula [here](https://github.com/Hasnep/homebrew-tap/blob/main/Formula/meta-package-manager.rb). I just followed [this guide](https://github.com/simonw/til/blob/main/homebrew/packaging-python-cli-for-homebrew.md), so it's very basic. You can try installing it with
```shell
brew install hasnep/tap/meta-package-manager
# Or alternatively
mpm --brew install hasnep/tap/meta-package-manager
```
I'm not 100% sure exactly what needs to happen next, the guide I followed said once it's ready it can be submitted to [homebrew-core](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md). I'll try that another day once I've tested it more thoroughly.

I might look into creating a deb package as well once the homebrew stuff is done because I mainly use Ubutnu, although I think that's a little bit harder.

**Comment by kdeldycke** [comment]
Thanks @Hasnep for your fantastic contribution! I just document your proposed installation with brew at: https://github.com/kdeldycke/meta-package-manager/commit/523907226c1366494fa3879d80585e355b6fa493

**Comment by kdeldycke** [comment]
Good point @autinerd! Just added a complete benchmark of distributions at: https://kdeldycke.github.io/meta-package-manager/benchmark.html#distribution

**Comment by Hasnep** [comment]
I force pushed to my tap repo and because I'm new to homebrew I didn't realise this would break it. If anyone's brew is broken it should be as simple as running `brew untap hasnep/tap` and then `brew install hasnep/tap/meta-package-manager`. Sorry! :sweat_smile:

**Comment by kdeldycke** [comment]
> I force pushed to my tap repo and because I'm new to homebrew I didn't realise this would break it. If anyone's brew is broken it should be as simple as running `brew untap hasnep/tap` and then `brew install hasnep/tap/meta-package-manager`. Sorry! 😅

Thanks @Hasnep for the heads up! I just added a [note in the documentation about this situation](https://github.com/kdeldycke/meta-package-manager/commit/fb6166264cfdac85527c7095a800bccbdcbc59bc).

**Comment by kdeldycke** [comment]
The distribution of `mpm` is tracked in the documentation at: https://kdeldycke.github.io/meta-package-manager/benchmark.html#distribution

So there is no need to let this issue open. We'll simply open a new issue or PR to track integration with new integration channels.
